src/components/ChatEntry.js

```diff
@@ -1,22 +1,51 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp.js';
+// import { useState } from 'react';
 
 const ChatEntry = (props) => {
+  // const [isLiked, setIsLiked] = useState(false);
+
+  // const toggleLiked = () => {
+  //   console.log('I have clicked on a heart!');
+  //   setIsLiked(!isLiked);
+  // };
+
+  const onLikeClick = () => {
+    console.log('I clicked a heart!');
+    const updatedChat = {
+      liked: !props.liked,
+    };
+    props.onUpdate(updatedChat);
+  };
+
+  const heartStatus = props.liked ? '❤️' : '🤍';
+
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+
+        <p>
+          <TimeStamp time={props.timeStamp} />
+        </p>
+        <button onClick={onLikeClick} className="heart-liked">
+          {heartStatus}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  onupdate: PropTypes.func.isRequired,
```

Comment:
Looks like there is a typo here! The `u` is uppercase everywhere else except here.